calico/felix/fetcd.py

```diff
@@ -132,6 +149,67 @@ def _periodically_resync(self):
             gevent.sleep(sleep_time)
             self.force_resync(reason="periodic resync", async=True)
 
+    @logging_exceptions
+    def _status_reporting(self):
+        """
+        Greenlet: periodically writes to etcd
+
+        :return: Does not return, unless reporting disabled.
+        """
+        _log.info("Started status reporting thread.")
+        ttl = self._config.REPORTING_TTL_SECS
+        interval = self._config.REPORTING_INTERVAL_SECS
+        _log.debug("Reporting interval: %s, TTL: %s", interval, ttl)
+
+        if interval == 0:
+            _log.info("Interval is 0, status reporting disabled.")
+            return
+
+        while True:
+            try:
+                self.update_felix_status(ttl)
+                gevent.sleep(interval)
+            except EtcdException as e:
```

Comment:
This except block cannot be hit because you pass `async=True` to the `write_to_etcd` method; which will make it return an `AsyncResult` instead of raising the exception.  that means that any exceptions will take down the process instead.

Also, there are other exceptions that need to be dealt with.  The etcd client may raise a read or connection timeout, for example.